### PR TITLE
Role backend implementation

### LINKED ISSUE #435: Role Backend Implementation
When fetching resources for a Module, some resources are supported to return details about a specific resource. For example, the endpoint for Deployments will return more information about pods and their statuses, and the endpoint for Services will return information about the ports.

If you want to fetch information about a Deployment from the default namespace called demo-app you would call:

```bash
/resources?group=apps&version=v1&kind=Deployment&name=demo-app&namespace=default
```

The goal of the issue is to support Role on the same endpoint. The endpoint for fetching Role data would look like this: 

```bash
 /resources?group=apps&version=v1&kind=Role&name=&namespace=
```

The handler function for the endpoint is [here](https://github.com/cyclops-ui/cyclops/blob/main/cyclops-ctrl/internal/controller/modules.go), and you will have to handle a new type [here](https://github.com/cyclops-ui/cyclops/blob/main/cyclops-ctrl/internal/cluster/k8sclient/client.go).

The endpoint should return data about its `name`, `namespace`, `rules`, `resources`, `verbs`… (you can find more about it [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.30/#role-v1-rbac-authorization-k8s-io).

Refer to this PR on adding new types https://github.com/cyclops-ui/cyclops/pull/259

### PR DESCRIPTION
**Pull request by tanbirali**
closes #435 

## 📑 Description
This PR supports the fetching information about a Role
The new types are added with ref to the PR #259 

## ✅ Checks
- [ ] I have updated the documentation as required
- [ ] I have performed a self-review of my code

### CODE REVIEW

File: cyclops-ctrl/internal/controller/modules.go [hunk 1]
```diff
@@ -653,3 +653,24 @@ func getTargetGeneration(generation string, module *v1alpha1.Module) (*v1alpha1.
...
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{
+			"error":  "Failed to fetch resource",
+			"reason": err.Error(),
+		})
+		return
+	}
+
+	ctx.JSON(http.StatusOK, resource)
+}
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/controller/modules.go:676]: We don't need this handler function. `GetResource` handler func will take care of it

File: cyclops-ctrl/internal/cluster/k8sclient/client.go [hunk 2]
```diff
@@ -877,6 +879,23 @@ func (k *KubernetesClient) isResourceNamespaced(gvk schema.GroupVersionKind) (bo
 	return false, errors.New(fmt.Sprintf("group version kind not found: %v", gvk.String()))
 }
 
+func (k *KubernetesClient) mapRole(group, version, kind, name, namespace string ) (*dto.Role, error){
+	 roleList, err := k.clientset.RbacV1().Roles(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	 if err != nil {
+		return nil, err
+	 }
+
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/cluster/k8sclient/client.go:887]: Can you leave a single empty line here?

File: cyclops-ctrl/internal/cluster/k8sclient/client.go [hunk 3]
```diff
@@ -877,6 +879,23 @@ func (k *KubernetesClient) isResourceNamespaced(gvk schema.GroupVersionKind) (bo
 	return false, errors.New(fmt.Sprintf("group version kind not found: %v", gvk.String()))
 }
 
+func (k *KubernetesClient) mapRole(group, version, kind, name, namespace string ) (*dto.Role, error){
+	 roleList, err := k.clientset.RbacV1().Roles(namespace).Get(context.Background(), name, metav1.GetOptions{})
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/cluster/k8sclient/client.go:883]: Can you rename this to role?

**Review by petar-cvit — CHANGES_REQUESTED** [review]

**Comment by tanbirali** [comment]
Made the requested changes, do have a look.

File: cyclops-ctrl/internal/cluster/k8sclient/client.go [hunk 4]
```diff
@@ -877,6 +879,24 @@ func (k *KubernetesClient) isResourceNamespaced(gvk schema.GroupVersionKind) (bo
 	return false, errors.New(fmt.Sprintf("group version kind not found: %v", gvk.String()))
 }
 
+func (k *KubernetesClient) role(group, version, kind, name, namespace string ) (*dto.Role, error){
+	 roleList, err := k.clientset.RbacV1().Roles(namespace).Get(context.Background(), name, metav1.GetOptions{})
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/cluster/k8sclient/client.go:883]: @tanbirali can you remove the variable not the func name

**Review by petar-cvit — CHANGES_REQUESTED** [review]

File: cyclops-ctrl/internal/cluster/k8sclient/client.go (hunk 4 above)
Comment (tanbirali) [reply cyclops-ctrl/internal/cluster/k8sclient/client.go]: Do you mean the roleList variable?

File: cyclops-ctrl/internal/cluster/k8sclient/client.go [hunk 5]
```diff
@@ -877,6 +879,24 @@ func (k *KubernetesClient) isResourceNamespaced(gvk schema.GroupVersionKind) (bo
 	return false, errors.New(fmt.Sprintf("group version kind not found: %v", gvk.String()))
 }
 
+func (k *KubernetesClient) mapRole(group, version, kind, name, namespace string ) (*dto.Role, error){
+	 role, err := k.clientset.RbacV1().Roles(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	 
+	 if err != nil {
+		return nil, err
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/cluster/k8sclient/client.go:886]: Make sure to format the code with `go fmt ./...`

**Review by petar-cvit — CHANGES_REQUESTED** [review]
You can add role rules to the object. `mapRole` function doesn't add any Role specific info to the response

File: cyclops-ctrl/internal/cluster/k8sclient/client.go (hunk 5 above)
Comment (tanbirali) [reply cyclops-ctrl/internal/cluster/k8sclient/client.go:886]: Doesn't the commit command formats the code, I have seen it do something but I am not sure if it formatting process

Comment (petar-cvit) [reply cyclops-ctrl/internal/cluster/k8sclient/client.go:886]: We have it in place only for the UI

**Review by petar-cvit — COMMENTED** [review]
@tanbirali can you run `go fmt ./...` from `cyclops-ctrl`

File: cyclops-ctrl/internal/cluster/k8sclient/client.go [hunk 6]
```diff
@@ -916,3 +935,7 @@ func isSecret(group, version, kind string) bool {
 func isCronJob(group, version, kind string) bool {
 	return group == "batch" && version == "v1" && kind == "CronJob"
 }
+
+func isRole(group, version, kind string) bool {
+	return group == "" && version == "v1" && kind == "Role"
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/cluster/k8sclient/client.go:940]: can you update the group here to `rbac.authorization.k8s.io`?

**Review by petar-cvit — CHANGES_REQUESTED** [review]
@tanbirali, left one more comment, but rest of the PR looks good 😄

File: cyclops-ctrl/internal/cluster/k8sclient/client.go (hunk 6 above)
Comment (tanbirali) [reply cyclops-ctrl/internal/cluster/k8sclient/client.go]: Yeah sure

**Review by petar-cvit — APPROVED** [review]
Thanks @tanbirali 🧡

**Pull request merged**